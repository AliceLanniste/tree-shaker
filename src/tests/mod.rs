use insta::{assert_snapshot, glob};
use std::fs;

fn tree_shake(input: &str) -> String {
  let result = crate::tree_shake(input);
  result.codegen_return.source_text
}

#[test]
fn test() {
  glob!("fixtures/**/*.js", |path| {
    let input = fs::read_to_string(path).unwrap();
    assert_snapshot!(tree_shake(&input));
  });
}
