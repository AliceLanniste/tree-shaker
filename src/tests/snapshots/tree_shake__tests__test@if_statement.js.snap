---
source: src/tests/mod.rs
expression: tree_shake(&input)
input_file: src/tests/fixtures/if_statement.js
---
export function main() {
	return 1;
}
